projects/packages/jetpack-mu-wpcom/src/features/wpcom-sidebar-notice/wpcom-sidebar-notice.php

```diff
@@ -33,53 +35,17 @@ function wpcom_enqueue_sidebar_notice_assets() {
 		'wpcom-sidebar-notice',
 		plugins_url( 'build/wpcom-sidebar-notice/wpcom-sidebar-notice.css', Jetpack_Mu_Wpcom::BASE_FILE ),
 		array(),
-		Jetpack_Mu_Wpcom::PACKAGE_VERSION
+		$asset_file['version'] ?? filemtime( Jetpack_Mu_Wpcom::BASE_DIR . 'build/wpcom-sidebar-notice/wpcom-sidebar-notice.css' )
 	);
 
-	$notice = wpcom_get_sidebar_notice();
-	if ( $notice ) {
-		$link = $notice['link'];
-		if ( str_starts_with( $link, '/' ) ) {
-			$link = 'https://wordpress.com' . $link;
-		}
-
-		$user_id    = null;
-		$user_login = null;
-
-		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-			global $current_user;
-			$user_id    = $current_user->ID;
-			$user_login = $current_user->user_login;
-		} else {
-			$connection_manager = new Connection_Manager();
-			$wpcom_user_data    = $connection_manager->get_connected_user_data();
-			if ( $wpcom_user_data ) {
-				$user_id    = $wpcom_user_data['ID'];
-				$user_login = $wpcom_user_data['login'];
-			}
-		}
-
-		$data = array(
-			'url'          => esc_url( $link ),
-			'text'         => wp_kses( $notice['content'], array() ),
-			'action'       => wp_kses( $notice['cta'], array() ),
-			'dismissible'  => $notice['dismissible'],
-			'dismissLabel' => esc_html__( 'Dismiss', 'jetpack-mu-wpcom' ),
-			'id'           => $notice['id'],
-			'featureClass' => $notice['feature_class'],
-			'dismissNonce' => wp_create_nonce( 'wpcom_dismiss_sidebar_notice' ),
-			'tracks'       => $notice['tracks'],
-			'user'         => array(
-				'ID'       => $user_id,
-				'username' => $user_login,
-			),
-		);
-
-		wp_add_inline_script(
-			'wpcom-sidebar-notice',
-			'window.wpcomSidebarNotice = ' . wp_json_encode( $data ) . ';'
-		);
-	}
+	wp_localize_script(
```

Comment:
Nitpick (non-blocking): `wp_localize_script` shouldn't be used to inject data from PHP to JS. See https://developer.wordpress.org/reference/functions/wp_localize_script/

> Though localization is the primary use, it was often used to pass generic data from PHP to JavaScript, because it was originally the only official way to do that. [wp_add_inline_script()](https://developer.wordpress.org/reference/functions/wp_add_inline_script/) was [introduced in WordPress Version 4.5](https://make.wordpress.org/core/2016/03/08/enhanced-script-loader-in-wordpress-4-5/), and is now the best practice for that use case. `[wp_localize_script()](https://developer.wordpress.org/reference/functions/wp_localize_script/) ` should only be used when you actually want to localize strings.